llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8306,8 +8313,9 @@ VPWidenCallRecipe *VPRecipeBuilder::tryToWidenCall(CallInst *CI,
                 },
                 Range);
   if (ShouldUseVectorIntrinsic)
-    return new VPWidenCallRecipe(CI, make_range(Ops.begin(), Ops.end()), ID,
-                                 CI->getDebugLoc());
+    return new VPWidenIntrinsicRecipe(*CI, ID,
+                                      make_range(Ops.begin(), Ops.end() - 1),
```

Comment:
Maybe move Ops.push_back(Operands.back()); from line 8305 to line 8319 to avoid -1 here?

Reply:
Updated, thanks!